### PR TITLE
Move getColumn to DataAccessHelpers

### PR DESCRIPTION
**Pull request by devinrsmith**
Partial #3692

### CODE REVIEW

File: Plot/src/main/java/io/deephaven/plot/util/tables/ColumnHandlerFactory.java [hunk 1]
```diff
@@ -117,7 +118,7 @@ private ColumnHandlerTable(final Table table, final String columnName, Class typ
 
         protected DataColumn getDataColumn() {
             if (dataColumn == null) {
-                dataColumn = table.getColumn(columnName);
+                dataColumn = ((BaseTable) table).getColumn(columnName);
```
Comment (rcaudy) [inline Plot/src/main/java/io/deephaven/plot/util/tables/ColumnHandlerFactory.java:121]: These will all be raw types warnings. Want to do a search replace to `(BaseTable<?>)`?

**Comment by rcaudy** [comment]
Please file a follow-up issue to fully delete `DataColumn` and offer convenient `toArray` on column iterators.

**Comment by devinrsmith** [comment]
Follow-up https://github.com/deephaven/deephaven-core/issues/3920

File: engine/table/src/main/java/io/deephaven/engine/table/impl/QueryTable.java [hunk 2]
```diff
@@ -262,9 +262,6 @@ default SwapListener newSwapListener(final QueryTable queryTable) {
 
     private static final AtomicReferenceFieldUpdater<QueryTable, ModifiedColumnSet> MODIFIED_COLUMN_SET_UPDATER =
             AtomicReferenceFieldUpdater.newUpdater(QueryTable.class, ModifiedColumnSet.class, "modifiedColumnSet");
-
-    private static final AtomicReferenceFieldUpdater<QueryTable, Map> INDEXED_DATA_COLUMNS_UPDATER =
-            AtomicReferenceFieldUpdater.newUpdater(QueryTable.class, Map.class, "indexedDataColumns");
     private static final Map<String, IndexedDataColumn<?>> EMPTY_INDEXED_DATA_COLUMNS = Collections.emptyMap();
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/QueryTable.java]: Delete this please

File: engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java [hunk 3]
```diff
@@ -28,6 +28,7 @@
 import io.deephaven.internal.log.LoggerFactory;
 import io.deephaven.io.log.impl.LogOutputStringImpl;
 import io.deephaven.io.logger.Logger;
+import io.deephaven.util.annotations.FinalDefault;
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java:31]: Intentional?

File: engine/table/src/main/java/io/deephaven/engine/table/impl/DataAccessHelpers.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@Deprecated
+public class DataAccessHelpers {
+
+    // -----------------------------------------------------------------------------------------------------------------
+    // DataColumns for fetching data by row position; generally much less efficient than ColumnSource
+    // TODO(deephaven-core#3920): Delete DataColumn
+    // -----------------------------------------------------------------------------------------------------------------
+
+    public static DataColumn<?>[] getColumns(Table table) {
+        return table.getDefinition().getColumnStream().map(c -> getColumn(table, c.getName()))
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/DataAccessHelpers.java:16]: You should coalesce here first.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/DataAccessHelpers.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+                .toArray(DataColumn[]::new);
+    }
+
+    public static <T> DataColumn<T> getColumn(Table table, final int columnIndex) {
+        return getColumn(table, table.getDefinition().getColumns().get(columnIndex).getName());
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/DataAccessHelpers.java:21]: And here

File: engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java (hunk 3 above)
Comment (devinrsmith) [reply engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java:31]: Removed

**Review by rcaudy — DISMISSED** [review]

**Review by rcaudy — APPROVED** [review]

**Pull request merged**